index.js

```diff
@@ -71,6 +78,17 @@ class Piece {
     return this.board.getSquareContent(row, file);
   }
 
+  isValidSquare(row, file) {
+    return (
+      row >= 0 && row < Board.LANE_SIZE && file >= 0 && file < Board.LANE_SIZE
+    );
+  }
+
+  isSquareOccupied(row, file) {
+    let squareContent = this.getSquareContent(row, file);
+    return squareContent !== null;
+  }
```

Comment:
I agree this is a better place for these. An even better place would be on the `Board` class because these methods handle concerns more closely related to the `Board`.
```suggestion
  isValidSquare(row, file) {
    return this.board.isValidSquare(row, file)
  }

  isSquareOccupied(row, file) {
    return this.board.isSquareOccupied(row, file)
  }
```